### ISSUE TITLE
Enhancement Suggestion, Calculate Range with Grid - Range Penalties based on In-Universe Measurements of Grid Tiles (E.G. 2 Yards/6 Feet per Inch, SWADE RAW)

### ISSUE THREAD

**Issue by tech097** [javierriveracastro/betteroll-swade#661]
After adjusting the Grid Scale of the map my session in Foundry for SWADE, I was greeted with the issue of having the Range Penalties of weapons being treated as much more strict then they were supposed to be.

<img width="1059" alt="Screenshot 2023-12-02 at 9 24 25 PM" src="https://github.com/javierriveracastro/betteroll-swade/assets/136868340/acb4e176-133e-4cf4-8452-df4a26e3b19b">
<img width="1059" alt="Screenshot 2023-12-02 at 9 25 07 PM" src="https://github.com/javierriveracastro/betteroll-swade/assets/136868340/98dc8b38-d97e-4290-8a8d-ef79cbfe9162">

As seen above, at just two tiles away from the enemy shot with a flintlock pistol, when the grid measures at 6 Units per Grid Tile, BR2's Range Penalty Tracker treats it like it's "Two Groups of 6 1-Unit-Sized Tiles", IE, the exact same thing as twelve 1-Unit tiles away from the enemy.

Because of this, I'm unable to list the in-universe measurements of a grid tile as 6 Feet. The only way for range penalties to be fairly calculated is by solely using 1-Unit/Inch Measurements for the tiles. However, I actually enjoy getting to build up immersion by having real-world macro-sized measurements used in the ruler function of Foundry, it adds an element of realism that I think helps add to the charm of it.

That said, I was curious if it would ever be possible to include a function in BR2 that allows the module to understand that that a grid scale that exceeds 1-Unit-Sized tiles doesn't necessarily mean that the bigger Unit-Sized tiles represent multiple tiles at once. That way, having tiles that are listed as 6 Units/Feet in Size, only treats them as Tiles that happen to be that size, not increments of 6 1-Unit/Inch Tiles cumulatively.

I'd think this would be a fun addition, but by all means the overall progress of BR2 should be focused on first and foremost, whether it's possible or not.

**Comment by javierriveracastro** [comment]
First, try using 5 feet grids. They should work as there is some already some logic to handle  D&D maps.

The main problem is that there is no concept of "grid" on the SWADE rules, so there need to be a place where you give and equivalence between maps units and inches. And it should be different for each map, so a global setting want work.

TL;DR: This issue needs more work to make it actionable, IMHO.

**Comment by TheLaslo** [comment]
This is not a problem with Better Rolls.  This is a problem with changing the range of the grid.  All items in the SWADE Core Rules Module and probably the SWADE System have ranges based on grid inches.  1 grid = 1 inch = 6'.  If you look at the flintlock pistol the ranges are 5/10/20.  If you want to change all the grid units in your scenes to 1 grid = 6' then you will have to change the ranges on all your weapon ranges by a factor of 6.. so 30/60/120 in your flintlock example.

**Comment by javierriveracastro** [comment]
I will close this, as it looks a lot like it is abandoned.
